src/api/cow-subgraph/index.ts

```diff
@@ -0,0 +1,97 @@
+import log from 'loglevel'
+import fetch from 'cross-fetch'
+import { GraphQLClient, gql, Variables } from 'graphql-request'
+import { CowError } from '/utils/common'
+import { SupportedChainId as ChainId } from '/constants/chains'
+import { Context } from '/utils/context'
+import { LastDaysVolumeQuery, LastHoursVolumeQuery, TotalsQuery } from './graphql'
+
+export function getSubgraphUrls(): Record<ChainId, string> {
+  return {
+    [ChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/gnosis/cow',
+    [ChainId.RINKEBY]: 'https://api.thegraph.com/subgraphs/name/gnosis/cow-rinkeby',
+    [ChainId.GNOSIS_CHAIN]: 'https://api.thegraph.com/subgraphs/name/gnosis/cow-gc',
+  }
+}
+
+export class CowSubgraphApi<T extends ChainId> {
+  chainId: T
+  context: Context
+  client: GraphQLClient
+
+  API_NAME = 'CoW Protocol Subgraph'
+
+  constructor(chainId: T, context: Context) {
+    this.chainId = chainId
+    this.context = context
+    this.client = new GraphQLClient(this.API_BASE_URL, { fetch: fetch })
+  }
+
+  get API_BASE_URL(): string {
+    return getSubgraphUrls()[this.chainId]
+  }
+
+  async getTotals(): Promise<TotalsQuery> {
+    log.debug(`[subgraph:${this.API_NAME}] Get totals for:`, this.chainId)
+    const query = gql`
+      query Totals {
```

Comment:
Should queries be exported to a `queries.ts` file ?

Reply:
As there are not many queries and the file is not that big, having them next to the place where they are consumed seems easier to follow for now